src/hotspot/share/classfile/protectionDomainCache.cpp

```diff
@@ -128,108 +159,69 @@ void ProtectionDomainCacheTable::unlink() {
 
   // Purge any deleted entries outside of the SystemDictionary_lock.
   purge_deleted_entries();
+  int oops_removed = purge_entries_from_table();
```

Comment:
Maybe add comment

`int oops_removed = purge_entries_from_table(); // reacquires SD lock`

otherwise it gives the false impression this is done lock-free.

Reply:
Ok.